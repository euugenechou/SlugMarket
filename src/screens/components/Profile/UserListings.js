/* React imports */
import React, { Component } from "react";
import { View, Text, StyleSheet, Image } from "react-native";

class UserListings extends Component {
  render() {
    return (
      <View
        style={{
          flexDirection: "row",
          width: this.props.width - 40,
          height: this.props.width / 2 - 80,
          borderWidth: 0.5,
          borderColor: "#dddddd",
          marginLeft: 20,
          marginTop: 20
        }}
      >
        <View style={{ flex: 2 }}>
          <Image
            style={{
              flex: 2,
              width: null,
              height: null,
              resizeMode: "cover"
            }}
            source={require("../../../assets/textbooks.jpg")}
          />
        </View>
        <View
          style={{
            flex: 1,
            alignItems: "center",
            justifyContent: "space-evenly"
          }}
        >
          <Text
            style={{ alignItems: "center", fontSize: 18, fontWeight: "bold" }}
          >
            {this.props.name}
          </Text>
          <Text style={{ alignItems: "center", fontSize: 18 }}>
            ${this.props.price}
          </Text>
        </View>
      </View>
    );
  }
}

export default UserListings;

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    justifyContent: "center"
  }
});
