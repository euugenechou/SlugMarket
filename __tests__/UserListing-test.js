// __tests__/UserListings-test.js
import 'react-native';
import React from 'react';
import UserListings from '../src/screens/components/Profile/UserListings';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <UserListings />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});