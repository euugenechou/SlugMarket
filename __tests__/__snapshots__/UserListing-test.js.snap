// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#dddddd",
      "borderRadius": 3,
      "borderWidth": 1,
      "elevation": 1,
      "flexDirection": "row",
      "height": NaN,
      "marginBottom": 20,
      "marginLeft": 20,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "width": NaN,
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/textbooks.jpg",
        }
      }
      style={
        Object {
          "borderRadius": 3,
          "flex": 2,
          "height": null,
          "resizeMode": "cover",
          "width": null,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignItems": "center",
          "fontSize": 14,
          "fontWeight": "bold",
          "justifyContent": "center",
          "paddingHorizontal": 10,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignItems": "center",
          "fontSize": 12,
          "justifyContent": "center",
          "paddingTop": 5,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignItems": "center",
          "fontSize": 12,
          "justifyContent": "center",
          "paddingTop": 5,
        }
      }
    >
      $
    </Text>
  </View>
</View>
`;
